docs/getting-started/tutorials/auth-architecture/external-idp-internal-session.md

```diff
@@ -0,0 +1,210 @@
+          "scope": [
+            "openid"
+          ],
+          "order": 10,
+          "userSettingsURL": "https://{{KEYCLOAK_URL}}/auth/realms/<realm>/account/"
+        }
+      },
+      "redirectUrl": "https://<MY-HOST>/web-login/oauth/callback",
+      "isWebsiteApp": true,
+      "realm": "<realm>",
+      "issuer": "<the signer of the jwt>",
+      "defaultGroups": []
+    }
+  }
+}
+```
+
+Here you can notice some important values to set:
+- `redirectUrl` must be `https://<MY-HOST>/web-login/oauth/callback`. This relative path will be used to handle the callback from the IdP during the Oauth2 login flow, and there we will expose the logic to complete the authentication flow.
+  - If you are using Auth0 as IdP and so `auth0-client` as your service, the path you need to set is `https://<MY-HOST>/web-login/callback`. The difference lies in the way the web page requests the token: with `/oauth` in the path will pass the info via the body of the request, without it will pass the info via the query parameters.
+- `isWebsiteApp` must be `true` if using the authentication for a web application, since it will make the Authentication Service return the session as a cookie.
+
+#### Authorization service
+Follow the [tutorial of the 'external IdP' scenario](/getting-started/tutorials/auth-architecture/external-idp.md#tutorial-steps).
+
+#### Login Site
+-- TODO
+
+### Exposing the auth flow APIs
+
+You have to create the following APIs:
+- `/web-login` [no auth] --> point to `login site`
+- `/authorize` [no auth] --> point to `authentication-service`
+- `/oauth/token` [no auth] --> point to `authentication-service`
+- `/logout` --> point to `authentication-service`
+- `/userinfo` (optional) --> point to `authorization-service`
+
+:::warning
+The first three have to be authentication free, otherwise the setup will not work.
+:::
+
+### Setup the flow using the secure-apigateway application
+To speed up the process, you can use the [secure-apigateway application](/runtime_suite_applications/secure-apigateway/overview.md), modifying it a little after the creation.
+
+This is because the application uses Auth0 as a default IdP, so:
+- if you want to use Auth0, then the application is ready to use, you don't need to do anything else. Follow the application doc for more.
+- if you want to use another Oauth2 compliant IdP:
+  - start by creating the `crud-service` and the `authentication-service` as above
+  - install the application, modifying the `auth0-client` with the `authentication-service` in the wizard
+  - change the environment variables of the `authorization-service` following the [tutorial of the 'external IdP' scenario](/getting-started/tutorials/auth-architecture/external-idp.md#tutorial-steps)
+  - remove the unnecessary environment variables specific of Auth0
+  - the application will take care of creating:
+    - api-gateway
+    - authorization-service
+    - login-site
+    - all the endpoints
+
+## How does it work?
```

Comment:
Maybe is better to move this section between the Definition and Tutorial steps sections? In that way you introduce first the interactions between the components and then you implement the solution. Of course this section would need a little rework to make it work.

Reply:
@EdoCrippaOfficial the `How does it work?` section overlaps with the `Definition` section. Indeed, the sequence diagram of the `Definition` section and all the images of the `How does it work?` section contain mostly the same information. Can you try to merge the two sections picking the content that you prefer?

Reply:
I removed the sequence diagram that would otherwise overlap with the "how it works" section.